kubectl-plugin/pipeline/gc.go

```diff
@@ -74,7 +88,7 @@ func (o *gcOption) cleanPipelineRunInNamespace(namespace string) (err error) {
 			break
 		}
 
-		if okToDelete(item.Object, o.maxAge) {
+		if (o.condition == conditionAnd && okToDelete(item.Object, o.maxAge)) || o.condition == conditionOr {
```

Comment:
I don't understand this logic here.

Reply:
I think it's clear. Only when the condition is `and` and it's time to delete

Or the condition is `or`

Reply:
Please consider this scenario: total of PipelineRuns is `10`, the `max-count` is equal to `1000`, `max-age` is equal to `1s` and `condition` is equal to `or`.

Reply:
Good point. But there's a case we need to consider. Let assume, there are no new PipelineRuns over 7 days. The GC will remove all PipelineRuns. It is not reasonable, because people still want to have a PipelineRun list. The key point of the GC is to save storage instead of clean it.

So I've removed the `or` condition, and add `ignoreTime` as well.